src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -354,12 +354,33 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	if (!stmt->is_grant)
 	{
 		DefElem *opt = NULL;
+		int opt_count = 0 ;
 		foreach_ptr(opt, stmt->opt)
 		{
-			if (strcmp(opt->defname, "admin") == 0)
+			switch (opt->defname)
 			{
-				appendStringInfo(buf, "ADMIN OPTION FOR ");
-				break;
+				case "admin":
+					appendStringInfo(buf, "ADMIN OPTION FOR ");
+					opt_count++;
```

Comment:
I think we cannot have multiple revoke options in one statement, can you also double check?
```SQL
$ revoke admin option for, inherit option for role1 from role2;
ERROR:  syntax error at or near ","
$ revoke admin option for role1 from role2, inherit option for role1 from role2;
ERROR:  syntax error at or near "option"
$ revoke admin, inherit option for role1 from role2;
ERROR:  syntax error at or near "option"
LINE 1: revoke admin, inherit option for role1 from role2;
```
If this is true, we don't need something like `opt_count` here.